pkg/nmi/server/server.go

```diff
@@ -346,3 +362,21 @@ func copyHeader(dst, src http.Header) {
 		}
 	}
 }
+
+func handleTermination() {
+	log.Info("Received SIGTERM, shutting down")
+
+	exitCode := 0
+	// clean up iptables
+	if err := iptables.DeleteCustomChain(); err != nil {
+		log.Infof("Error cleaning up during shutdown: %v", err)
+		exitCode = 1
+	}
+
+	// wait for pod to delete
```

Comment:
Which pod is this wait for ?

Reply:
This is just wait for the graceful shutdown of the pod. This is not absolutely necessary as the termination grace period is 60s. So we will kill the container before the termination grace period.